src/legacy/ui/public/saved_objects/components/saved_object_save_modal.tsx

```diff
@@ -16,56 +16,148 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
-import React, { Fragment } from 'react';
-import PropTypes from 'prop-types';
-import { FormattedMessage } from '@kbn/i18n/react';
-
 import {
   EuiButton,
+  EuiCallOut,
   EuiFieldText,
+  EuiForm,
+  EuiFormRow,
   EuiModal,
   EuiModalBody,
   EuiModalFooter,
   EuiModalHeader,
   EuiModalHeaderTitle,
   EuiOverlayMask,
   EuiSpacer,
-  EuiCallOut,
-  EuiForm,
-  EuiFormRow,
   EuiSwitch,
 } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React, { Fragment } from 'react';
 
-export class SavedObjectSaveModal extends React.Component {
-  constructor(props) {
-    super(props);
+interface OnSaveProps {
+  newTitle: string;
+  newCopyOnSave: boolean;
+  isTitleDuplicateConfirmed: boolean;
+  onTitleDuplicate: () => void;
+}
 
-    this.state = {
-      title: props.title,
-      copyOnSave: false,
-      isTitleDuplicateConfirmed: false,
-      hasTitleDuplicate: false,
-      isLoading: false,
-    };
-  }
-  componentDidMount() {
-    this._isMounted = true;
-  }
+interface Props {
+  onSave: (arg0: OnSaveProps) => void;
```

Comment:
Let's rename `arg0` to `props` for a bit more saying name.

Reply:
Ok!